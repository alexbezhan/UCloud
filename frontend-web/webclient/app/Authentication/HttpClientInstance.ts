import HttpClient from "./lib";
import {WebSocketFactory} from "./ws";

export const Client = new HttpClient();
export const WSFactory = new WebSocketFactory(Client);
