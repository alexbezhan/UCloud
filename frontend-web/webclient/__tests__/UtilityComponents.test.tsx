import {configure, shallow} from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import * as React from "react";
import {Provider} from "react-redux";
import {create} from "react-test-renderer";
import {ThemeProvider} from "styled-components";
import {Client} from "../app/Authentication/HttpClientInstance";
import {dialogStore} from "../app/Dialog/DialogStore";
import {SortBy, SortOrder} from "../app/Files";
import {theme} from "../app/ui-components";
import {mockFile} from "../app/Utilities/FileUtilities";
import {store} from "../app/Utilities/ReduxUtilities";
import {
    addStandardDialog,
    Arrow,
    FileIcon,
    overwriteDialog,
    PP,
    rewritePolicyDialog,
    sensitivityDialog,
    shareDialog,
    SharePrompt
} from "../app/UtilityComponents";
import {iconFromFilePath} from "../app/UtilityFunctions";

configure({adapter: new Adapter()});

describe("PP", () => {

    test("Non-visible PP-Component", () => {
        expect(create(<PP visible={false} />).toJSON()).toMatchSnapshot();
    });

    test("Visible PP-Component", () => {
        expect(create(<PP visible />).toJSON()).toMatchSnapshot();
    });

    test.skip("Change PP-value", () => {
        const pP = shallow(<PP visible />);
        pP.findWhere(it => !!it.props().type().range).simulate("change", {target: {value: "500"}});
        expect((pP.state() as {duration: number}).duration).toBe(500);
    });
});

describe("FileIcon", () => {

    test("FileIcon, not shared", () => {
        const mFile = mockFile({path: "path", type: "DIRECTORY"});
        const iconType = iconFromFilePath(mFile.path, mFile.fileType);
        expect(create(
            <ThemeProvider theme={theme}>
                <FileIcon fileIcon={iconType} />
            </ThemeProvider>
        )).toMatchSnapshot();
    });
    test("FileIcon, shared", () => {
        const mFile = mockFile({path: "path", type: "DIRECTORY"});
        const iconType = iconFromFilePath(mFile.path, mFile.fileType);
        expect(create(
            <ThemeProvider theme={theme}>
                <FileIcon
                    fileIcon={iconType}
                    shared
                />
            </ThemeProvider>
        )).toMatchSnapshot();
    });
});

describe("Arrow", () => {
    test("arrowUp", () => {
        expect(create(
            <Arrow activeSortBy={SortBy.PATH} sortBy={SortBy.PATH} order={SortOrder.ASCENDING} />
        )).toMatchSnapshot();
    });

    test("arrowDown", () => {
        expect(create(
            <Arrow activeSortBy={SortBy.PATH} sortBy={SortBy.PATH} order={SortOrder.DESCENDING} />
        )).toMatchSnapshot();
    });

    test("undefined", () => {
        expect(create(
            <Arrow activeSortBy={SortBy.PATH} sortBy={SortBy.FILE_TYPE} order={SortOrder.ASCENDING} />)
        ).toMatchSnapshot();
    });
});


describe("Dialogs", () => {
    test("Add standard dialog", () => {
        let dialogCount = 0;
        dialogStore.subscribe(dialogs => dialogCount = dialogs.length);
        addStandardDialog({
            title: "Title",
            message: "Message",
            onConfirm: () => undefined
        });
        expect(dialogCount).toBe(1);
        dialogStore.failure();
        expect(dialogCount).toBe(0);
    });

    test("Add sensitivity dialog", () => {
        let dialogCount = 0;
        dialogStore.subscribe(dialogs => dialogCount = dialogs.length);
        sensitivityDialog();
        expect(dialogCount).toBe(1);
        dialogStore.failure();
        expect(dialogCount).toBe(0);
    });

    test("Add share dialog", () => {
        let dialogCount = 0;
        dialogStore.subscribe(dialogs => dialogCount = dialogs.length);
        shareDialog([], Client);
        expect(dialogCount).toBe(1);
        dialogStore.failure();
        expect(dialogCount).toBe(0);
    });

    test("Add overwrite dialog", () => {
        let dialogCount = 0;
        dialogStore.subscribe(dialogs => dialogCount = dialogs.length);
        overwriteDialog();
        expect(dialogCount).toBe(1);
        dialogStore.failure();
        expect(dialogCount).toBe(0);
    });

    test("Add rewritePolicy dialog, no overwrite", () => {
        let dialogCount = 0;
        dialogStore.subscribe(dialogs => dialogCount = dialogs.length);
        rewritePolicyDialog({
            allowOverwrite: false,
            filesRemaining: 0,
            client: Client,
            path: "path",
            projects: []
        });
        expect(dialogCount).toBe(1);
        dialogStore.failure();
        expect(dialogCount).toBe(0);
    });

    test("Add rewritePolicy dialog, no overwrite", () => {
        let dialogCount = 0;
        dialogStore.subscribe(dialogs => dialogCount = dialogs.length);
        rewritePolicyDialog({
            allowOverwrite: true,
            filesRemaining: 0,
            client: Client,
            path: "path",
            projects: []
        });
        expect(dialogCount).toBe(1);
        dialogStore.failure();
        expect(dialogCount).toBe(0);
    });
});

test("Share prompt", () => {
    expect(create(
        <Provider store={store}>
            <ThemeProvider theme={theme}>
                <SharePrompt paths={[""]} client={Client} />
            </ThemeProvider>
        </Provider>
    ).toJSON()).toMatchSnapshot();
});
