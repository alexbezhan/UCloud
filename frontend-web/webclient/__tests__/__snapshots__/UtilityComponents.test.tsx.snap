// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow arrowDown 1`] = `
.c0 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: pointer;
  margin-right: .4em;
}

<svg
  className="c0"
  clipRule="evenodd"
  cursor="pointer"
  fill="currentcolor"
  fillRule="evenodd"
  height=".7em"
  mr=".4em"
  viewBox="0 0 18 24"
  width=".7em"
>
  <path
    d="M9 24l9-10h-6V0H6v14h-6l9 10z"
  />
</svg>
`;

exports[`Arrow arrowUp 1`] = `
.c0 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: pointer;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  margin-right: .4em;
}

<svg
  className="c0"
  clipRule="evenodd"
  cursor="pointer"
  fill="currentcolor"
  fillRule="evenodd"
  height=".7em"
  mr=".4em"
  rotation="180"
  viewBox="0 0 18 24"
  width=".7em"
>
  <path
    d="M9 24l9-10h-6V0H6v14h-6l9 10z"
  />
</svg>
`;

exports[`Arrow undefined 1`] = `null`;

exports[`FileIcon FileIcon, not shared 1`] = `
.c0 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: inherit;
  color: #8393A7;
}

<svg
  className="c0"
  clipRule="evenodd"
  color="FtFolderColor"
  color2="#c9d3df"
  cursor="inherit"
  fill="currentcolor"
  fillRule="evenodd"
  height={30}
  viewBox="0 0 25 23"
  width={30}
>
  <path
    d="M0 1.5A1.503 1.503 0 011.5 0H9l3 4h10.5A1.5 1.5 0 0124 5.5V8H0V1.5z"
    fill="#c9d3df"
  />
  <path
    d="M0 7.5A1.5 1.5 0 011.5 6h21A1.5 1.5 0 0124 7.5l.001 13.5a1.002 1.002 0 01-1 1H1a1 1 0 01-1-1V7.5z"
  />
</svg>
`;

exports[`FileIcon FileIcon, shared 1`] = `
.c2 {
  cursor: auto;
  position: absolute;
  bottom: -6px;
  right: -2px;
}

.c6 {
  cursor: inherit;
  font-size: 14px;
}

.c1 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: inherit;
  color: #8393A7;
}

.c4 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: inherit;
}

.c3 {
  position: relative;
  display: inline-block;
}

.c3:hover > div {
  display: block;
}

.c5 {
  overflow-y: auto;
  overflow-x: hidden;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,.14),0px 1px 18px 0px rgba(0,0,0,.12);
  display: none;
  position: absolute;
  background-color: #f5f7f9;
  color: #1e252e;
  width: 160px;
  min-width: 138pxpx;
  padding: 12px 16px;
  z-index: 47;
  text-align: left;
  cursor: pointer;
  opacity: 0;
  pointer-events: none;
}

.c5 svg {
  margin-right: 1em;
}

.c5 > svg ~ span {
  margin-right: 1em;
}

.c0 {
  cursor: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

<div
  className="c0"
  cursor="inherit"
>
  <svg
    className="c1"
    clipRule="evenodd"
    color="FtFolderColor"
    color2="#c9d3df"
    cursor="inherit"
    fill="currentcolor"
    fillRule="evenodd"
    height={30}
    viewBox="0 0 25 23"
    width={30}
  >
    <path
      d="M0 1.5A1.503 1.503 0 011.5 0H9l3 4h10.5A1.5 1.5 0 0124 5.5V8H0V1.5z"
      fill="#c9d3df"
    />
    <path
      d="M0 7.5A1.5 1.5 0 011.5 6h21A1.5 1.5 0 0124 7.5l.001 13.5a1.002 1.002 0 01-1 1H1a1 1 0 01-1-1V7.5z"
    />
  </svg>
  <div
    className="c2"
    cursor="auto"
  >
    <div
      className="c3"
    >
      <svg
        className="c4"
        clipRule="evenodd"
        color2="#fff"
        cursor="inherit"
        fill="currentcolor"
        fillRule="evenodd"
        height="15px"
        viewBox="0 0 34 28"
        width="15px"
      >
        <path
          d="M30.564 13.817L16.778 25.679v-7.325h-.09l-.08.013-.08.012-.08.013h-.081l-.077.012-.167.025-.166.025-.166.038-.168.025-.168.037-.17.038-.169.037-.17.05-.172.038-.172.05-.173.05-.172.05-.174.05-.173.062-.173.063-.173.062-.173.063-.173.062-.172.063-.172.075-.17.075-.171.062-.171.088-.168.075-.168.075-.169.075-.165.087-.165.088-.163.087-.163.075-.16.088-.159.1-.157.087-.156.088-.154.1-.152.087-.15.1-.148.088-.144.1-.143.1-.141.1-.138.1-.136.1-.134.1-.13.1-.126.1-.125.1-.122.1-.117.1-.116.1-.11.1-.109.1-.105.1-.1.1-.099.1-3.874 4.05.336-5.575.012-.15.013-.15.016-.15.02-.15.02-.15.024-.138.026-.15.028-.15.03-.137.032-.15.035-.15.036-.138.038-.15.042-.137.042-.138.045-.15.046-.137.048-.138.05-.137.054-.138.053-.137.056-.138.058-.137.06-.138.06-.137.064-.138.065-.137.066-.125.067-.138.07-.125.07-.137.074-.125.074-.125.076-.138.077-.125.08-.125.08-.125.082-.125.083-.125.085-.125.086-.125.087-.125.089-.112.09-.125.091-.113.094-.125.093-.112.096-.125.096-.113.099-.112.1-.113.099-.112.1-.113.104-.112.103-.113.106-.1.106-.112.107-.113.109-.1.11-.1.11-.112.111-.1.113-.1.114-.1.114-.1.116-.1.117-.1.117-.088.12-.1.12-.1.12-.087.121-.088.124-.1.123-.087.123-.088.126-.087.126-.088.127-.075.128-.087.129-.088.13-.075.13-.075.13-.087.133-.075.132-.075.133-.075.135-.075.135-.063.135-.075.137-.062.137-.075.138-.063.138-.062.14-.063.14-.062.139-.063.142-.05.142-.062.142-.05.143-.05.144-.05.144-.05.145-.05.145-.05.146-.05.147-.038.147-.037.147-.05.148-.025.148-.038.15-.037.15-.038.149-.025.15-.025.151-.025.151-.025.152-.025.152-.025.152-.012.153-.025.153-.013.154-.012h.016V1.954l13.786 11.863zm-11.787-2.5h-1.496l-.126.012h-.124l-.126.013-.126.012-.124.013-.126.012-.126.013-.125.012-.125.025-.125.013-.126.025-.124.025-.125.025-.126.025-.124.025-.124.037-.124.025-.125.038-.123.037-.123.038-.124.037-.123.038-.123.037-.123.05-.122.038-.122.05-.12.037-.121.05-.121.05-.121.05-.12.05-.12.063-.119.05-.118.062-.118.05-.118.063-.117.062-.117.063-.116.062-.114.063-.116.062-.113.063-.115.075-.112.062-.113.075-.111.075-.111.063-.111.075-.11.075-.11.075-.106.075-.109.087-.106.075-.106.075-.105.088-.104.075-.104.087-.103.088-.102.075-.1.087-.1.088-.099.087-.098.088-.097.087-.096.1-.096.088-.093.087-.094.1-.092.088-.092.1-.09.087-.088.1-.089.1-.087.1-.085.1-.084.1-.084.088-.082.1-.081.112-.08.1-.08.1-.077.1-.076.1-.075.113-.073.1-.072.112-.071.1-.07.113-.069.1-.066.112-.066.1-.064.113-.063.112-.06.113-.06.1-.059.112-.057.113-.055.112-.054.113-.052.112-.051.113-.05.112-.047.113-.046.112-.045.113-.043.112-.04.113-.04.112-.04.125-.035.113-.034.112-.033.113-.032.112-.028.125-.028.113-.026.112-.024.113-.022.125-.02.112-.019.113-.016.112-.015.125-.012.113-.011.112-.01.125-.007.113.116-.125.12-.113.122-.125.126-.125.13-.112.131-.125.135-.113.138-.125.142-.112.143-.113.147-.125.15-.112.152-.113.154-.112.157-.113.158-.112.162-.113.165-.1.166-.112.168-.1.17-.113.171-.1.174-.1.176-.1.177-.1.18-.1.18-.1.182-.1.183-.087.186-.088.185-.087.187-.1.19-.075.189-.088.19-.087.19-.075.193-.075.193-.075.193-.075.194-.075.196-.075.194-.063.196-.062.195-.063.197-.062.196-.05.196-.05.196-.063.197-.037.195-.05.196-.05.195-.038.196-.037.194-.025.195-.038.194-.025.096-.012.096-.013.097-.012.096-.013.096-.012.095-.013h.095l.095-.012h.096l.095-.013h.188l.093-.012h1.526v5l8.72-7.5-8.72-7.5v5z"
          fill="#fff"
        />
        <path
          d="M27.497 13.817l-8.72 7.5v-5h-1.246c-3.96 0-8.777 2.412-11.21 5 .257-4.963 5.873-10 11.21-10h1.246v-5l8.72 7.5z"
        />
      </svg>
      <div
        className="c5"
        color="text"
        cursor="pointer"
        disabled={false}
        width="160px"
      >
        <div
          className="c6"
          cursor="inherit"
          fontSize={1}
        >
          This file is shared
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PP Non-visible PP-Component 1`] = `null`;

exports[`PP Visible PP-Component 1`] = `
.c0 {
  direction: rtl;
}

<div
  className="center"
>
  <svg
    stroke="#000"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="3"
    viewBox="0 0 128 128"
    x="0px"
    y="0px"
  >
    <path
      id="body"
    >
      <animate
        attributeName="fill"
        dur="500ms"
        keyTimes="0;0.1;0.2;0.3;0.4;0.5;0.6;0.7;0.8;0.9;1"
        repeatCount="indefinite"
        values="
                                #ff8d8b;
                                #fed689;
                                #88ff89;
                                #87ffff;
                                #8bb5fe;
                                #d78cff;
                                #ff8cff;
                                #ff68f7;
                                #fe6cb7;
                                #ff6968;
                                #ff8d8b
                            "
      />
      <animate
        attributeName="d"
        dur="500ms"
        keyTimes="0;0.1;0.2;0.3;0.4;0.5;0.6;0.7;0.8;0.9;1"
        repeatCount="indefinite"
        values="
                                M67.1,109.5c-9.6,0-23.6-8.8-23.6-24c0-12.1,17.8-41,37.5-41c15.6,0,23.3,10.6,25.1,20.5 c1.9,10.7,3.9,8.2,3.9,19.5c0,8.2-3.8,17-3.8,22.3c0,3.9,1.4,7.4,2.9,10.5c1.7,3.5,2.4,6.6,2.4,9.2H9.5c0-13.7,10.8-14,21.1-23 c5.6-4.9,11-12.4,14.5-26;
                                M56.1,107.5c-9.6,0-24.6-13.8-24.6-29c0-16.2,13-42,33.5-42c17.8,0,22.3,11.6,26.1,22.5 c3.6,10.3,9.9,9.2,9.9,20.5c0,8.2-1.8,7-1.8,12.3c0,4.5,3.4,8.2,6.4,14.1c2.5,4.8,4.8,11.2,4.8,20.6h-99c0-12.1,7.2-17.6,14.7-24.3 c3.2-2.9,6.5-5.9,9.2-9.8;
                                M45.1,109.5c-5.5-0.2-27.6-8.4-27.6-27c0-17.9,14.8-42,32.5-42c15.4,0,24,10.4,26.1,21.5 c1.3,6.7,9.9,9.8,9.9,21.5c0,8.2-0.8,6-0.8,11.3c0,7.7,12.8,9,20.8,15c7.7,5.8,15.5,16.7,15.5,16.7h-110c0-4.8,1.7-11.3,5-16 c3.2-4.5,4.5-8.3,5-15;
                                M36,120c-5.5-0.2-28.5-11.9-28.5-30.5c0-16.2,12.5-42,33-42c17.8,0,21.8,9.6,25.6,20.5 C69.7,78.3,76,78.2,76,89.5c0,8.2-0.8,4-0.8,9.3c0,5.9,16.5,7.8,28.4,15.9c8,5.5,17.9,11.8,17.9,11.8h-110c0-2.1-1.2-5.2-1.9-14.5 c-0.3-3.6-0.5-8.1-0.5-13.9;
                                M37,119.5c-15,0.1-33.5-12.7-33.5-30C3.5,73.3,16,47,36.5,47c17.8,0,22.8,11,26,22 C65.6,79.4,73,79.2,73,90.5c0,4-1.8,6.6-1.8,8.3c0,5.9,14.2,6.4,26.4,15.9c7.7,6,13.9,11.8,13.9,11.8H7.5c-1.2-3.4-1.8-7.3-1.9-11.2 c-0.2-5.1,0.3-10.1,0.9-13.7;
                                M40.5,121.5c-12.6,0-30-13.4-30-29C10.5,76.3,23,53,43.5,53c14.5,0,22.8,9.6,25,22 c1.2,6.9,10,9.2,10,20.5c0,4,0,5.6,0,7.3c0,4.9,6.1,7.5,11.2,11.9c5.8,5,7.2,11.8,7.2,11.8H8.5c0-1.5-0.6-6.1,0.4-11.8 c0.6-3.5,1.9-7.5,4.3-11.6;
                                M48.5,121.5c-12.6,0-25-6.3-25-18c0-16.2,13.7-47,36-47c15.6,0,24.8,9.1,27,21.5 c1.2,6.9,7,9.2,7,18.5c0,9.5-4,11-4,22c0,4.1,0.5,5,1,6c0.5,1.2,1,2,1,2h-81c0-5.3,3.1-8.3,6.3-11.5c2.6-2.6,5.4-5.3,6.7-9.5;
                                M68.5,121.5c-12.6,0-33-5.8-33-23c0-9.2,11.8-36,37-36c15.6,0,25.8,8.1,28,20.5 c1.2,6.9,4,6.2,4,15.5c0,9.5-5,15.1-5,21c0,1.9,1,2.3,1,5c0,1.2,0,2,0,2h-91c0.5-7.6,7.1-11.1,13.7-15.7c4.9-3.4,9.9-7.5,12.3-14.3;
                                M73.5,117.5c-12.6,0-30-6.2-30-25c0-14.2,20.9-37,38-37c17.6,0,25.8,11.1,28,23.5 c1.2,6.9,3,7.2,3,16.5c0,12.1-6,16.1-6,22c0,4.2,2,5.3,2,8c0,1.2,0,1,0,1H7.5c2.1-9.4,10.4-13.3,19.2-19.4 c7.1-5,14.4-11.5,18.8-23.6;
                                M80.5,115.5c-12.6,0-32-9.2-32-28c0-14.2,22.9-35,40-35c17.6,0,25.8,12.1,28,24.5 c1.2,6.9,3,6.2,3,15.5c0,12.1-6,19.1-6,25c0,4.2,2,5.3,2,8c0,1.2,0,1,0,1h-102c2.3-8.7,11.6-11.7,20.8-20.1 c5.3-4.8,10.5-11.4,14.2-21.9;
                                M67.1,109.5c-9.6,0-23.6-8.8-23.6-24c0-12.1,17.8-41,37.5-41c15.6,0,23.3,10.6,25.1,20.5 c1.9,10.7,3.9,8.2,3.9,19.5c0,8.2-3.8,17-3.8,22.3c0,3.9,1.4,7.4,2.9,10.5c1.7,3.5,2.4,6.6,2.4,9.2H9.5c0-13.7,10.8-14,21.1-23 c5.6-4.9,11-12.4,14.5-26
                            "
      />
    </path>
    <path
      fill="#7b8c68"
      id="beak"
    >
      <animate
        attributeName="d"
        dur="500ms"
        keyTimes="0;0.1;0.2;0.3;0.4;0.5;0.6;0.7;0.8;0.9;1"
        repeatCount="indefinite"
        values="
                                M78.29,70c0-9.92,2.5-14,8-14s8,2.17,8,10.67c0,15.92-7,26.33-7,26.33S78.29,85.5,78.29,70Z;
                                M62.29,64c0-9.92,2.5-14,8-14s8,2.17,8,10.67c0,15.92-7,26.33-7,26.33S62.29,79.5,62.29,64Z;
                                M48.29,67c0-9.92,2.5-14,8-14s8,2.17,8,10.67c0,15.92-7,26.33-7,26.33S48.29,82.5,48.29,67Z;
                                M36.29,73c0-9.92,2.5-14,8-14s8,2.17,8,10.67c0,15.92-7,26.33-7,26.33S36.29,88.5,36.29,73Z;
                                M35.29,75c0-9.92,2.5-14,8-14s8,2.17,8,10.67c0,15.92-7,26.33-7,26.33S35.29,90.5,35.29,75Z;
                                M41.29,81c0-9.92,2.5-14,8-14s8,2.17,8,10.67c0,15.92-7,26.33-7,26.33S41.29,96.5,41.29,81Z;
                                M59.29,84c0-9.92,2.5-14,8-14s8,2.17,8,10.67c0,15.92-7,26.33-7,26.33S59.29,99.5,59.29,84Z;
                                M72.29,89c0-9.92,2.5-14,8-14s8,2.17,8,10.67c0,15.92-7,26.33-7,26.33S72.29,104.5,72.29,89Z;
                                M80.29,82c0-9.92,2.5-14,8-14s8,2.17,8,10.67c0,15.92-7,26.33-7,26.33S80.29,97.5,80.29,82Z;
                                M87.29,78c0-9.92,2.5-14,8-14s8,2.17,8,10.67c0,15.92-7,26.33-7,26.33S87.29,93.5,87.29,78Z;
                                M78.29,70c0-9.92,2.5-14,8-14s8,2.17,8,10.67c0,15.92-7,26.33-7,26.33S78.29,85.5,78.29,70Z"
      />
    </path>
    <ellipse
      id="eye-right"
      rx="3"
      ry="4"
    >
      <animate
        attributeName="cx"
        dur="500ms"
        keyTimes="0;0.1;0.2;0.3;0.4;0.5;0.6;0.7;0.8;0.9;1"
        repeatCount="indefinite"
        values="100;84;70;58;57;63;81;94;102;109;100"
      />
      <animate
        attributeName="cy"
        dur="500ms"
        keyTimes="0;0.1;0.2;0.3;0.4;0.5;0.6;0.7;0.8;0.9;1"
        repeatCount="indefinite"
        values="62;56;59;65;67;73;76;81;74;70;62"
      />
    </ellipse>
    <ellipse
      id="eye-left"
      rx="3"
      ry="4"
    >
      <animate
        attributeName="cx"
        dur="500ms"
        keyTimes="0;0.1;0.2;0.3;0.4;0.5;0.6;0.7;0.8;0.9;1"
        repeatCount="indefinite"
        values="67.5;51.5;37.5;25.5;24.5;30.5;48.5;61.5;69.5;76.5;67.5"
      />
      <animate
        attributeName="cy"
        dur="500ms"
        keyTimes="0;0.1;0.2;0.3;0.4;0.5;0.6;0.7;0.8;0.9;1"
        repeatCount="indefinite"
        values="62;56;59;65;67;73;76;81;74;70;62"
      />
    </ellipse>
  </svg>
  <input
    className="c0"
    id="animationDuration"
    max="2000"
    min="200"
    onChange={[Function]}
    step="1"
    type="range"
    value={500}
  />
</div>
`;

exports[`Share prompt 1`] = `
.c5 {
  cursor: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 8px;
}

.c16 {
  cursor: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 15px;
}

.c0 {
  cursor: auto;
  width: 620px;
  max-height: 80vh;
}

.c1 {
  cursor: auto;
  width: 605px;
}

.c12 {
  cursor: pointer;
  margin-left: -17px;
  padding-left: 15px;
  margin-right: -17px;
  width: auto;
}

.c15 {
  cursor: auto;
}

.c17 {
  cursor: auto;
  margin-right: auto;
}

.c19 {
  cursor: auto;
  margin-left: auto;
}

.c2 {
  cursor: inherit;
  font-size: 24px;
  margin: 0;
  font-weight: 400;
}

.c9 {
  cursor: pointer;
}

.c10 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: inherit;
  margin-left: .7em;
}

.c14 {
  font-smoothing: antialiased;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 5px;
  background-color: #006aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  -webkit-transition: cubic-bezier(0.5,0,0.25,1) 60ms;
  transition: cubic-bezier(0.5,0,0.25,1) 60ms;
  font-size: 14px;
  padding: 9.5px 18px;
  margin-left: 5px;
}

.c14:disabled {
  opacity: 0.25;
}

.c14:focus {
  outline: none;
}

.c14:hover {
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

.c18 {
  font-smoothing: antialiased;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 5px;
  background-color: #006aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  -webkit-transition: cubic-bezier(0.5,0,0.25,1) 60ms;
  transition: cubic-bezier(0.5,0,0.25,1) 60ms;
  font-size: 14px;
  padding: 9.5px 18px;
}

.c18:disabled {
  opacity: 0.25;
}

.c18:focus {
  outline: none;
}

.c18:hover {
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

.c3 {
  border: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  margin-left: 0;
  margin-right: 0;
  border-color: #c9d3df;
}

.c6 {
  display: block;
  font-family: inherit;
  color: inherit;
  background-color: transparent;
  margin: 0;
  border-width: 2px;
  border-color: #c9d3df;
  border-style: solid;
  padding-top: 7px;
  padding-bottom: 7px;
  padding-left: 12px;
  padding-right: 12px;
  border-radius: 5px;
  width: 100%;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.c6::-webkit-input-placeholder {
  color: #8393A7;
}

.c6::-moz-placeholder {
  color: #8393A7;
}

.c6:-ms-input-placeholder {
  color: #8393A7;
}

.c6::placeholder {
  color: #8393A7;
}

.c6:focus {
  outline: none;
  background-color: transparent;
}

.c6:focus {
  outline: 0;
  border-color: #006aff;
}

.c7 {
  cursor: inherit;
  width: 152px;
  border: #c9d3df solid 2px;
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
  border-left: 0px;
  margin-left: 0;
  width: 152px;
  padding-left: 1em;
  padding-right: 1em;
  padding-top: 6px;
}

.c4 {
  font-size: 10px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  display: block;
  margin: 0;
  font-size: 14px;
  color: #000;
  font-weight: 700;
  width: 100%;
}

.c8 {
  position: relative;
  display: inline-block;
}

.c8:hover > div {
  display: block;
}

.c11 {
  overflow: visible;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,.14),0px 1px 18px 0px rgba(0,0,0,.12);
  position: absolute;
  background-color: #fff;
  color: #000;
  width: 180px;
  min-width: 138pxpx;
  padding: 12px 16px;
  z-index: 47;
  text-align: left;
  cursor: pointer;
  opacity: 0;
  pointer-events: none;
}

.c11 > *:hover:not(.c13) {
  background-color: rgba(0,0,0,0.05);
}

.c11 svg {
  margin-right: 1em;
}

.c11 > svg ~ span {
  margin-right: 1em;
}

<div
  className="c0"
  cursor="auto"
  style={
    Object {
      "overflowY": "scroll",
    }
  }
  width="620px"
>
  <div
    className="c1"
    cursor="auto"
    width="605px"
  >
    <form
      onSubmit={[Function]}
    >
      <h3
        className="c2"
        cursor="inherit"
        fontSize={4}
      >
        Share
      </h3>
      <hr
        className="c3"
      />
      Collaborators
      <label
        className="c4"
        color="black"
        fontSize={1}
        fontWeight="bold"
        width="100%"
      >
        <div
          className="c5"
          cursor="inherit"
        >
          <input
            className="c6"
            id="default"
            placeholder="Enter username..."
            required={true}
            type="text"
            width="100%"
          />
          <div
            className="c7"
            cursor="inherit"
            width="152px"
          >
            <div
              className="c8"
              data-tag="dropdown"
            >
              <span
                className="c9"
                cursor="pointer"
                onClick={[Function]}
              >
                Can view
                <svg
                  className="c10"
                  clipRule="evenodd"
                  cursor="inherit"
                  fill="currentcolor"
                  fillRule="evenodd"
                  height=".7em"
                  ml=".7em"
                  viewBox="0 0 24 12"
                  width=".7em"
                >
                  <path
                    d="M0 0l11.959 12 12-12H0z"
                  />
                </svg>
              </span>
              <div
                className="c11"
                color="black"
                cursor="pointer"
                disabled={false}
                onClick={[Function]}
                overflow="visible"
                width="180px"
              >
                <div
                  className="c12"
                  cursor="pointer"
                  onClick={[Function]}
                  width="auto"
                >
                  Can view
                </div>
                <div
                  className="c12"
                  cursor="pointer"
                  onClick={[Function]}
                  width="auto"
                >
                  Can edit
                </div>
              </div>
            </div>
          </div>
          <button
            className="c13 c14"
            color="blue"
            onClick={[Function]}
          >
            Add
          </button>
        </div>
      </label>
      <div
        className="c15"
        cursor="auto"
      >
        No shares for 
        <b>
          
        </b>
      </div>
    </form>
    <div
      className="c16"
      cursor="inherit"
    >
      <div
        className="c17"
        cursor="auto"
      />
      <button
        className="c13 c18"
        color="blue"
        onClick={[Function]}
        type="button"
      >
        Close
      </button>
      <div
        className="c19"
        cursor="auto"
      />
    </div>
  </div>
</div>
`;
